common/lib/xmodule/xmodule/js/src/video/00_sjson.js

```diff
@@ -51,11 +67,38 @@ function() {
             return min;
         };
 
+        var filter = function (start, end) {
+            var filteredTimes = [],
+                filteredCaptions = [],
+                startTimes = getStartTimes(),
+                captions = getCaptions(),
+                cur,
+                i;
+
+            if (startTimes.length != captions.length) {
```

Comment:
makes sense, will update. I guess I'm used to using `!=` for length since the `length` property of arrays is very hard to override without encountering huge bugs.